near-plugins-derive/src/utils.rs

```diff
@@ -1,18 +1,58 @@
 use proc_macro2::Span;
 use proc_macro_crate::crate_name;
-use syn::{Ident, ItemFn};
+use std::str::FromStr;
+use syn::{FnArg, Ident, ItemFn};
 
-/// Determines if this block of code was generated by near_bindgen.
+/// Determines if this block of code was [generated by near_bindgen].
 /// Heuristic used is to check for #[no_mangle].
 /// TODO: How to make this 100% safe. Discuss with near-sdk team
+///
+/// [generated by near_bindgen]: https://github.com/near/near-sdk-rs/issues/722
 pub(crate) fn is_near_bindgen_wrapped_or_marshall(item: &ItemFn) -> bool {
-    let pattern1 = "(target_arch = \"wasm32\")";
-    let pattern2 = "(not(target_arch = \"wasm32\"))";
+    let condition_1 = {
+        let pattern1 = "(target_arch = \"wasm32\")";
+        let pattern2 = "(not(target_arch = \"wasm32\"))";
 
-    item.attrs.iter().any(|attr| {
-        let seq = attr.tokens.to_string();
-        seq == pattern1 || seq == pattern2
-    })
+        item.attrs.iter().any(|attr| {
+            let seq = attr.tokens.to_string();
+            seq == pattern1 || seq == pattern2
+        })
+    };
+    if condition_1 {
+        return true;
+    }
+
+    // For a struct `Contract`, `#[near-bindgen]` [generates] `ContractExt`. If
+    // `item` is in an implementation of `ContractExt`, the span number of its
+    // signature differs from the span number of the self token.
+    //
+    // [generates]: https://github.com/near/near-sdk-rs/pull/742
```

Comment:
An idea for a less hacky approach is asking `near-sdk` to _not_ forward some attributes. It could look like:

```rust
#[near_bindgen(dont_forward(access_control_any))]
impl Contract {
    // ...
}
```

Reply:
I like this approach. Shared on near-sdk-rs: https://github.com/near/near-sdk-rs/issues/722#issuecomment-1267651275